File: base/promena-transformer/promena-transformer-internal/src/test/java/pl/beone/promena/transformer/internal/model/data/TestUtils.java
package pl.beone.promena.transformer.internal.model.data;

import java.io.File;
import java.io.FileOutputStream;
import java.io.IOException;

class TestUtils {

    static File createTempFile(String string) {
        try {
            File file = File.createTempFile("tmp", null, null);
            try (FileOutputStream fileOutputStream = new FileOutputStream(file)) {
                fileOutputStream.write(string.getBytes());
            }
            return file;
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }

    @SuppressWarnings("ResultOfMethodCallIgnored")
    static File createTempDir() {
        try {
            File file = File.createTempFile("tmp", null, null);
            file.delete();
            file.mkdir();
            return file;
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: extension/communication/external/file/promena-communication-external-file-internal/src/test/kotlin/pl/beone/promena/communication/external/memory/internal/TestUtils.kt
package pl.beone.promena.communication.external.memory.internal

import pl.beone.promena.transformer.internal.model.data.FileData
import java.io.File
import java.net.URI

internal fun String.toFileData(location: URI): FileData =
        FileData(createTempFile(directory = File(location)).apply {
            writeText(this@toFileData)
        }.toURI())